src/modules/audio/audio.c

```diff
@@ -67,13 +70,19 @@ static bool mix(Source* source, float* output, uint32_t count) {
         // could skip min-ing with one of the buffers if you can guarantee that one is bigger/equal to the other (you can because their formats are known)
 
     uint64_t framesIn = source->sound->read(source->sound, source->offset, chunk, raw);
-    uint64_t framesOut = sizeof(aux) / (sizeof(float) * (2 >> source->spatial));
+    uint64_t framesOut = sizeof(aux) / (sizeof(float) * source->output_channel_count);
 
     ma_data_converter_process_pcm_frames(source->converter, raw, &framesIn, aux, &framesOut);
 
-    memcpy(mix, aux, framesOut * 2 * sizeof(float));
+    if(source->spatial) {
+      if(state.spatializer) {
```

Comment:
If the source is spatial but there isn't a spatializer, `mix` never gets written to, right?

Hm, it may be that the `spatial` flag needs to get forced to false for Sources if no spatializer successfully loads at initialization time.

Reply:
Style: ifs should have spaces after them

Reply:
Good call. Spatial forced to false if no spatializer is available.

Though, in my implementation there should always be a spatializer. If no third party one is available, the dummy one should work at least. There's a bug so that it's null in some cases but we can fix that; and my math in it is wrong so audio orientation is wrong when you turn; but with those fixes spatial audio will always be available even though it's not oculus/valve quality.

Reply:
Ooh yeah, if we can guarantee that there's a spatializer (even a dummy/null spatializer that does nothing), then we don't even need to worry about this.  Saves on branching too.

Reply:
Also, what's the story with the dummy spatializer?  Do you think it's good enough to be used as the default if one of the fancier spatializers isn't loaded?  Or was it just for testing here